File: src/main/java/net/flyingfishflash/loremlist/domain/lrmitem/LrmItemSuccinct.java
package net.flyingfishflash.loremlist.domain.lrmitem;

import java.util.UUID;
import net.flyingfishflash.loremlist.domain.SuccinctLrmComponent;

public record LrmItemSuccinct(UUID id, String name) implements SuccinctLrmComponent {

  public static LrmItemSuccinct fromLrmItem(LrmItem lrmItem) {
    return new LrmItemSuccinct(lrmItem.id(), lrmItem.name());
  }
}

File: src/main/kotlin/net/flyingfishflash/loremlist/domain/lrmitem/LrmItemSuccinct.kt
package net.flyingfishflash.loremlist.domain.lrmitem

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import net.flyingfishflash.loremlist.core.serialization.UuidSerializer
import net.flyingfishflash.loremlist.domain.SuccinctLrmComponent
import java.util.UUID

@Serializable
@SerialName("item")
data class LrmItemSuccinct(
  @Serializable(with = UuidSerializer::class)
  override val id: UUID,
  override val name: String,
) : SuccinctLrmComponent {
  companion object {
    fun fromLrmItem(lrmItem: LrmItem): LrmItemSuccinct {
      return LrmItemSuccinct(id = lrmItem.id, name = lrmItem.name)
    }
  }
}
